controllers/infinispan_config_controller.go

```diff
@@ -150,30 +152,38 @@ func (r configRequest) computeAndReconcileConfigMap(xsite *config.XSite) (*recon
 	}
 
 	jgroupsDiagnostics := consts.JGroupsDiagnosticsFlag == "TRUE"
-	serverConf := config.InfinispanConfiguration{
-		Infinispan: config.Infinispan{
-			Authorization: config.Authorization{
-				Enabled:    r.infinispan.IsAuthorizationEnabled(),
-				RoleMapper: roleMapper,
-			},
-			ClusterName: name,
-		},
-		JGroups: config.JGroups{
-			Transport: "tcp",
-			DNSPing: config.DNSPing{
-				Query: fmt.Sprintf("%s-ping.%s.svc.cluster.local", name, namespace),
-			},
-			Diagnostics: jgroupsDiagnostics,
-		},
-		Endpoints: config.Endpoints{
-			Authenticate:   r.infinispan.IsAuthenticationEnabled(),
-			DedicatedAdmin: true,
-		},
-		Logging: config.Logging{
-			Categories: r.infinispan.GetLogCategoriesForConfig(),
-		},
-	}
+	serverConf := config.InfinispanConfiguration{}
 
+	// serverConf config is populated with the defaultConfig.yaml ported from the config generator project
+	var defaultConfig string
+	if box, err := rice.FindBox("resources/"); err != nil {
+		return &reconcile.Result{}, err
+	} else {
+		if defaultConfig, err = box.String("defaultConfig.yaml"); err != nil {
+			return &reconcile.Result{}, err
+		}
+	}
+	if err := yaml.Unmarshal([]byte(defaultConfig), &serverConf); err != nil {
+		return &reconcile.Result{}, err
+	}
+	// The following are missing the default yaml (consider adding there?)
```

Comment:
Tbh, I think the cleanest and probably simplest solution is to remove the use of `*.xml`  entirely.

We can update `pkg/infinispan/configuration/configuration.go` to consist of structs that represent the yaml structure of the server and JGroups configuration [1]. That way, we can directly Marshall the struct to yaml and don't have to do any templating or use Rice.

The configuration struct doesn't have to be exhaustive, just the elements we require, so no need to define cache structs etc. Yaml configuration support in the server is pretty strong now and is used by our helm-charts, so I think it's fine to rely on this.

[1] Currently `jgroupsXmlTemplate.xmltmpl ` defines the JGroups relay stack directly in the JGroups xml format. We might have to make it so that this is defined in the `jgroups` element the same as the "image-tcp" stack.

Reply:
The approach was to port things from config controller as they was to reduce probs of errors. We can improve things anyway.
I would do a better analysis about removing templates:
- it seems to me we can easily apply server configuration format changes
- is somewhat a known format
- I didn't investigate deeply but I had problem with xml namespaces passing yaml as config

Reply:
> The approach was to port things from config controller as they were to reduce probs of errors.

That's true and it is always something that we can refactor later if necessary. When looking at the code though, it seemed simpler/cleaner to me to just make the change on the initial PR as it removes a lot of redundant code.

Reply:
> > The approach was to port things from config controller as they were to reduce probs of errors.
> 
> That's true and it is always something that we can refactor later if necessary. When looking at the code though, it seemed simpler/cleaner to me to just make the change on the initial PR as it removes a lot of redundant code.

limited to the default config I think it's doable on this pr (let's see if I'm able to CI turns stable green first). This changes will anyway turn the yaml into a big literal assignment with go syntax I suppose :)